weatherman.py

```diff
@@ -0,0 +1,177 @@
+import os
+import sys
+import argparse
+import datetime
+import glob
+import csv
+
+class DayReport:
+  def __str__(self):
+    print(self.date)
+
+  def __init__(self, date, maxtemp, meantemp, mintemp, dew, meandew, mindew, maxhumidity, meanhumidity, minhumidity, maxsealevel, meansealevel, minsealevel, maxvisibility, meanvisibility, minvisibility, maxwind, meanwind, maxgust, precipitaion, cloudcover, events, winddirdegrees):
+    self.date = date
+    self.maxtemp = maxtemp
+    self.meantemp = meantemp
+    self.mintemp = mintemp
+    self.dew = dew
+    self.meandew = meandew
+    self.mindew = mindew
+    self.maxhumidity = maxhumidity
+    self.meanhumidity = meanhumidity
+    self.minhumidity = minhumidity
+    self.maxsealevel = maxsealevel
+    self.meansealevel = meansealevel
+    self.minsealevel = minsealevel
+    self.maxvisibility = maxvisibility
+    self.meanvisibility = meanvisibility
+    self.minvisibility = minvisibility
+    self.maxwind = maxwind
+    self.meanwind = meanwind
+    self.maxgust = maxgust
+    self.precipitaion = precipitaion
+    self.cloudcover = cloudcover
+    self.events = events
+    self.winddirdegrees = winddirdegrees
+    
+  
+
+def populate_readings():
+  dayslist = []
+  dayscount = 0
+  path = sys.argv[1]
+  path += r"\*.txt"
+  files = glob.glob(path)
+  for csv_file in files:
+    with open(csv_file) as file:
+      reader = csv.DictReader(file, delimiter = ",")
+      line_count = 0
+      for row in reader:
+        if("PKT" in row):
+          l = row["PKT"].split('-')
+          d = datetime.date(int(l[0]), int(l[1]), int(l[2]))
+          dayslist.append(DayReport(d, row["Max TemperatureC"], row["Mean TemperatureC"], row["Min TemperatureC"], row["Dew PointC"], row["MeanDew PointC"], row["Min DewpointC"], row["Max Humidity"], row[" Mean Humidity"], row[" Min Humidity"], row[" Max Sea Level PressurehPa"], row[" Mean Sea Level PressurehPa"], row[" Min Sea Level PressurehPa"], row[" Max VisibilityKm"], row[" Mean VisibilityKm"], row[" Min VisibilitykM"], row[" Max Wind SpeedKm/h"], row[" Mean Wind SpeedKm/h"], row[" Max Gust SpeedKm/h"], row["Precipitationmm"], row[" CloudCover"], row[" Events"], row["WindDirDegrees"]))
+        else:
+          l1 = row["PKST"].split('-')
+          d1 = datetime.date(int(l1[0]), int(l1[1]), int(l1[2]))
+          dayslist.append(DayReport(d1, row["Max TemperatureC"], row["Mean TemperatureC"], row["Min TemperatureC"], row["Dew PointC"], row["MeanDew PointC"], row["Min DewpointC"], row["Max Humidity"], row[" Mean Humidity"], row[" Min Humidity"], row[" Max Sea Level PressurehPa"], row[" Mean Sea Level PressurehPa"], row[" Min Sea Level PressurehPa"], row[" Max VisibilityKm"], row[" Mean VisibilityKm"], row[" Min VisibilitykM"], row[" Max Wind SpeedKm/h"], row[" Mean Wind SpeedKm/h"], row[" Max Gust SpeedKm/h"], row["Precipitationmm"], row[" CloudCover"], row[" Events"], row["WindDirDegrees"]))
+        dayscount += 1
+        line_count += 1
```

Comment:
a lot of code is common in if/else section, follow [D.R.Y principle](https://www.geeksforgeeks.org/dont-repeat-yourselfdry-in-software-development/)